spec/lib/task-spec.js

```diff
@@ -508,6 +502,64 @@ describe("Task", function () {
 
     });
 
+    describe("timer", function() {
+        it("should not timeout when value is 0", function() {
+            var definition = _.cloneDeep(noopDefinition);
+            definition.options.timeout = 0;
+            var task = Task.create(definition, {}, {});
+            task.renderAll = sinon.stub().resolves();
+            sinon.spy(task, 'cancel');
+
+            return task.run().then(function() {
+                expect(task.state).to.equal('succeeded');
+                expect(task.error).to.equal(null);
+            });
+        });
+
+        it("should not timeout when value is -1", function() {
+            var definition = _.cloneDeep(noopDefinition);
+            definition.options.timeout = -1;
+            var task = Task.create(definition, {}, {});
+            task.renderAll = sinon.stub().resolves();
+            sinon.spy(task, 'cancel');
+
+            return task.run().then(function() {
+                expect(task.state).to.equal('succeeded');
+                expect(task.error).to.equal(null);
+            });
+        });
+
+        it("should timeout with options.timeout", function() {
+            var definition = _.cloneDeep(noopDefinition);
+            definition.options.timeout = 1;
+            var task = Task.create(definition, {}, {});
+            task.renderAll = sinon.stub().resolves();
+            sinon.spy(task, 'cancel');
+
+            return task.run().then(function() {
+                expect(task.state).to.equal('timeout');
+                expect(task.error).to.be.an.instanceof(Errors.TaskTimeoutError);
+                expect(task.error.message).to.equal("Task did not complete within 1ms");
+                expect(task.cancel).to.have.been.calledOnce;
+            });
+        });
+
+        it("should timeout with options.schedulerOverrides", function() {
+            var definition = _.cloneDeep(noopDefinition);
+            definition.options.schedulerOverrides = { timeout: 1 };
+            var task = Task.create(definition, {}, {});
+            task.renderAll = sinon.stub().resolves();
+            sinon.spy(task, 'cancel');
+
+            return task.run().then(function() {
+                expect(task.state).to.equal('timeout');
+                expect(task.error).to.be.an.instanceof(Errors.TaskTimeoutError);
+                expect(task.error.message).to.equal("Task did not complete within 1ms");
```

Comment:
@hohene I'm just hardcoding it in the test assertion against the error string, which will be generated via `{ timeout: 1 }` on line 549 (https://github.com/RackHD/on-tasks/pull/184#discussion-diff-62383049R549). In the non-test code it will produce the right message. Happy to change it here too though if you want.